jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +85,43 @@ def check_referer(self):
             return False
         return True
 
+    def get_scope_filter(self, req_scope):
+        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return.
+        Filter is a callable that takes a resource name and outputs true or false"""
+        kind_regex = re.compile(r':?(user|service|group)s:?')
+        try:
+            kind = re.search(kind_regex, req_scope).group(1)
+        except AttributeError:
+            self.log.warning(
+                "Regex error while processing scope %s, throwing 500", req_scope
+            )
+            raise web.HTTPError(
+                log_message="Unrecognized scope guard on method: %s" % req_scope
+            )
+        try:
+            sub_scope = self.parsed_scopes[req_scope]
+        except AttributeError:
+            raise web.HTTPError(
+                403,
+                "Resource scope %s (that was just accessed) not found in parsed scope model"
+                % req_scope,
+            )
+
+        def has_access(resource_name):
```

Comment:
I think it will be better to pass the actual orm object as `resource` rather than resource_name. That will make things easier. For example, `kind` can be based on the type of `resource` instead of parsing the scope with a regex.